### PR TITLE
Reallocate

### PR DESCRIPTION
**Pull request by giovanni-rosotti**
This monstruosity should implement the dynamical reallocation. I think this is now working, I had MANY crashes... I don't want do anything related to memory management in the near future!!!
The code seems valgrind clean now.

### CODE REVIEW

**Comment by giovanni-rosotti** [comment]
Now that this solves #79 and #80 we should be ready to accept it, let me know what you think

File: src/Hydrodynamics/Hydrodynamics.cpp [hunk 1]
```diff
@@ -202,26 +201,25 @@ void Hydrodynamics<ndim>::CheckYBoundaryGhostParticle
   const DomainBox<ndim> &simbox)       ///< [in] Simulation domain box
 {
   if (ndim > 1) {
-    Particle<ndim>& part = GetParticlePointer(i);
 
-    if (part.r[1] + min((FLOAT) 0.0, part.v[1]*tghost) <
-        simbox.boxmin[1] + ghost_range*kernrange*part.h) {
+    if (GetParticlePointer(i).r[1] + min((FLOAT) 0.0, GetParticlePointer(i).v[1]*tghost) <
```
Comment (rbooth200) [inline src/Hydrodynamics/Hydrodynamics.cpp]: I think this is ugly... Wouldn't it be easier to read if we just copied r and v instead?

File: src/Tree/Tree.cpp [hunk 2]
```diff
@@ -83,11 +83,17 @@ int Tree<ndim,ParticleType,TreeCell>::ComputeActiveParticleList
 //=================================================================================================
 template <int ndim, template<int> class ParticleType, template<int> class TreeCell>
 int Tree<ndim,ParticleType,TreeCell>::ComputeActiveCellList
- (TreeCell<ndim> *celllist)            ///< Array containing copies of cells with active ptcls
+ (TreeCell<ndim>** celllistaux)            ///< Array containing copies of cells with active ptcls
```
Comment (rbooth200) [inline src/Tree/Tree.cpp]: I'm not keen on a function that allocated memory and leaves someone else to clean up after it.  It's begging for someone to create a memory leak. Can we not instead use vectors here?

Comment (giovanni-rosotti) [inline src/Tree/Tree.cpp]: You've got a point, see 	06ce03c

File: src/Hydrodynamics/Hydrodynamics.cpp (hunk 1 above)
Comment (giovanni-rosotti) [inline src/Hydrodynamics/Hydrodynamics.cpp]: Fixed, see de177c2

**Pull request merged**